File: OOP_variable/3.py
class Cal(object):
    #생성자
    #2번째 arg가 a로 가, 그래서 맨 앞에 self를 추가
    def __init__(self,a,b):
        self.a=a
        self.b=b
    #메소드에서도 첫 번째 arg에는 self를 준다
    def add(self):
        return self.a+self.b
    def subtract(self):
        return self.a-self.b
    def setA(self, a):
        if isinstance(a,int):
            self.a=a
    def setB(self, b):
        if isinstance(b,int):
            self.b=b
    def getA(self):
        return self.a
    def getB(self):
        return self.b

#initialize method with name of class
c1 = Cal(10,10)
print(c1.add())
print(c1.subtract())

c1.setA(20)
c1.setB(30)
print(c1.add())
print(c1.subtract())

File: OOP_variable/3.rb
class Cal
  #생성자
  #인스턴스가 생성될 때 꼭 실행되어야 하는 메소드
  def initialize(a, b)
    @a = a  #인스턴스 변수 = 지역 변수
    @b = b
  end
  def add
    return @a+@b
  end
  def subtract
    return @a-@b
  end
  def setA a
    if a.is_a? Integer
      @a=a
    end
  end
  def setB b
    if b.is_a? Integer
      @b=b
    end
  end
  def getA
    return @a
  end
  def getB
    return @b
  end
end

#initialize method with 'new' method
c1 = Cal.new(10,10)
p c1.add
p c1.subtract
c1.setA 20
c1.setB 30
p c1.add
p c1.subtract
p c1.getB
